recommenders/utils/notebook_utils.py

```diff
@@ -35,3 +39,113 @@ def is_databricks():
             return False
     except NameError:
         return False
+
+
+def execute_notebook(
+    input_notebook, output_notebook, parameters={}, kernel_name="python3", timeout=600
+):
+    """Execute a notebook while passing parameters to it.
+
+    .. note::
+
+    Ensure your Jupyter Notebook is set up with parameters that can be
+    modified and read. Use Markdown cells to specify parameters that need
+    modification and code cells to set parameters that need to be read.
+
+    Args:
+        input_notebook (str): Path to the input notebook.
+        output_notebook (str): Path to the output notebook
+        parameters (dict): Dictionary of parameters to pass to the notebook.
+        kernel_name (str): Kernel name.
+        timeout (int): Timeout (in seconds) for each cell to execute.
+    """
+
+    # Load the Jupyter Notebook
+    with open(input_notebook, "r") as notebook_file:
+        notebook_content = nbformat.read(notebook_file, as_version=4)
+
+    # Search for and replace parameter values in code cells
+    for cell in notebook_content.cells:
+        if (
+            "tags" in cell.metadata
+            and "parameters" in cell.metadata["tags"]
+            and cell.cell_type == "code"
+        ):
+            cell_source = cell.source
+            modified_cell_source = (
+                cell_source  # Initialize a variable to hold the modified source
+            )
+            for param, new_value in parameters.items():
+                # Check if the new value is a string and surround it with quotes if necessary
+                if isinstance(new_value, str):
+                    new_value = f'"{new_value}"'
+                # Define a regular expression pattern to match parameter assignments and ignore comments
+                pattern = re.compile(
+                    rf"\b{param}\s*=\s*([^#\n]+)(?:#.*$)?", re.MULTILINE
+                )
+                matches = re.findall(pattern, cell_source)
+                for match in matches:
+                    old_assignment = match.strip()
+                    modified_cell_source = modified_cell_source.replace(
+                        old_assignment, f"{new_value}"
```

Comment:
just `str(new_value)` if we don't format any strings here

Reply:
Fixed in the next commit